core/db/db.go

```diff
@@ -172,32 +146,23 @@ func getDBPath() string {
 	return dbPath + "db"
 }
 
-func createDBIfDBPathEmpty() error {
-	dbPath := getDBPath()
-	missing, err := dirMissingOrEmpty(dbPath)
-	if err != nil {
-		return err
-	}
-	dbLogger.Debug("Is db path [%s] empty [%t]", dbPath, missing)
-	if missing {
-		err := CreateDB()
-		if err != nil {
-			return nil
-		}
+// Open open underlying rocksdb
+func (openchainDB *OpenchainDB) Open() {
+	openchainDB.mux.Lock()
+	if openchainDB.dbState == opened {
+		openchainDB.mux.Unlock()
+		return
 	}
-	return nil
-}
 
-func openDB() (*OpenchainDB, error) {
-	if isOpen {
-		return openchainDB, nil
-	}
+	defer openchainDB.mux.Unlock()
 
 	dbPath := getDBPath()
+	mkDirIfNotExist(dbPath)
+
 	opts := gorocksdb.NewDefaultOptions()
 	defer opts.Destroy()
 
-	opts.SetCreateIfMissing(false)
+	opts.SetCreateIfMissing(true)
```

Comment:
Earlier, I had two separate functions. 1) `createDB` and 2) `openDB`. I think that I should have documented the purpose. Now, you have merged the two into `open` function.
The purpose of separation was to handle a special case - suppose a user by mistake specifies the path of an existing 'non-empty' folder (as the path to be used for the db). In the earlier approach, he would have received an error (an we would have prevented rocksDB to produce it's file and polluting the folder). However, now, rocksDB would go ahead and start using the folder as db. Apart from polluting the existing user folder, I am not sure what other long term implication could be (e.g., does user continues to access other files he had? does rocksDB compaction mechanics ignore these files or does something special?...). I know that combining the two makes the code cleaner but thinking about above unusual scenario's I chose to be on a safer note and not to allow user to specify a folder for db that already has files/folders.